index.js

```diff
@@ -34,6 +33,28 @@ function load() {
   return result;
 }
 
+function isPlainObject(v) {
+  return typeof v === 'object' && v !== null && !Array.isArray(v);
+}
+
+function extend(target, source) {
+  if (!isPlainObject(target) || !isPlainObject(source)) {
+    throw new Error('Both target and source must be plain objects');
+  }
+
+  // iterate over all enumerable properties
+  for (const [key, value] of Object.entries(source)) {
+    // recursively extend if target has the same key, otherwise just assign
+    if (Object.prototype.hasOwnProperty.call(target, key)) {
+      target[key] = extend(extend({}, target[key]), value);
```

Comment:
```suggestion
      extend(target[key], value);
```

Leaning into the fact that `extend` modifies in-place.